flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java

```diff
@@ -150,6 +150,12 @@ public void testMerged() {
 		validate(parameter);
 	}
 
+	@Test
+	public void testFromGenericOptionsParser() throws IOException {
+		ParameterTool parameter = ParameterTool.fromGenericOptionsParser(new String[]{"-D", "input=myInput", "-DexpectedCount=15"});
+		validate(parameter);
+	}
```

Comment:
Sorry that I didn't write this earlier when I did the first review of the pull request:
Can you also add a test where we read one of the arguments (say input=myinput) from a xml configuration file.
As far as I can see, the GenericOptionsParser also supports specifying `-conf path/to/file.xml`.

That would be amazing ,)